ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -1994,8 +2014,614 @@ public void replicaTokenTest() throws InterruptedException {
     remoteReplicaInfo.onTokenPersisted();
   }
 
+  /**
+   * Tests when the local records has higher lifeVersion than remote records.
+   */
+  @Test
+  public void replicaThreadLifeVersionLocalGreaterThanRemote() throws Exception {
+    lifeVersionLocalGreaterThanRemote_Delete(false, false);
+    lifeVersionLocalGreaterThanRemote_Delete(false, true);
+    lifeVersionLocalGreaterThanRemote_Delete(true, false);
+    lifeVersionLocalGreaterThanRemote_Delete(true, true);
+  }
+
+  /**
+   * Tests when the local store missing put records with lifeVersion greater than 0
+   */
+  @Test
+  public void replicaThreadLifeVersionLocalLessThanRemote_MissingPuts() throws Exception {
```

Comment:
Is there any version test?
For example:

Local: x x x x D2
Remote: U2

U2 should be denied.

Reply:
There is, it's in another test method. `replicaThreadLifeVersionLocalLessThanRemote_FinalState_TtlUpdateDelete`